pkg/cephconfig/config.go

```diff
@@ -0,0 +1,397 @@
+	LongDesc           string      `json:"long_desc"`
+	Default            interface{} `json:"default"`
+	DaemonDefault      interface{} `json:"daemon_default"`
+	Tags               []string    `json:"tags"`
+	Services           []string    `json:"services"`
+	SeeAlso            []string    `json:"see_also"`
+	EnumValues         []string    `json:"enum_values"`
+	Min                interface{} `json:"min"`
+	Max                interface{} `json:"max"`
+	CanUpdateAtRuntime bool        `json:"can_update_at_runtime"`
+	Flags              []string    `json:"flags"`
+}
+
+// QueryParams contains the parameters for config search
+type QueryParams struct {
+	Service  *pb.ConfigParam_ServiceType
+	Level    *pb.ConfigParam_ConfigLevel
+	Name     *string
+	FullText *string
+	Sort     *pb.SearchConfigRequest_SortField
+	Order    *pb.SearchConfigRequest_SortOrder
+	Type     *pb.ConfigParam_ParamType
+}
+
+// ConfigParams is a slice of parameter information
+type ConfigParams []ConfigParamInfo
+
+// Config manages Ceph configuration parameters
+type Config struct {
+	params ConfigParams
+}
+
+// serviceTypeMap maps service enum values to their string representation
+var serviceTypeMap = map[pb.ConfigParam_ServiceType]string{
+	pb.ConfigParam_common:                 "common",
+	pb.ConfigParam_mon:                    "mon",
+	pb.ConfigParam_mds:                    "mds",
+	pb.ConfigParam_osd:                    "osd",
+	pb.ConfigParam_mgr:                    "mgr",
+	pb.ConfigParam_rgw:                    "rgw",
+	pb.ConfigParam_rbd:                    "rbd",
+	pb.ConfigParam_rbd_mirror:             "rbd-mirror",
+	pb.ConfigParam_immutable_object_cache: "immutable-object-cache",
+	pb.ConfigParam_mds_client:             "mds_client",
+	pb.ConfigParam_cephfs_mirror:          "cephfs-mirror",
+	pb.ConfigParam_ceph_exporter:          "ceph-exporter",
+}
+
+// ServiceStringToEnum maps service string to enum value
+var ServiceStringToEnum = func() map[string]pb.ConfigParam_ServiceType {
+	m := make(map[string]pb.ConfigParam_ServiceType)
+	for k, v := range serviceTypeMap {
+		m[v] = k
+	}
+	return m
+}()
+
+// loadParamsSlice loads all Ceph configuration params from the embedded JSON file into a sorted slice
+// NOTE: config-index.json should be sorted by the "name" field already
+func loadParamsSlice(ctx context.Context) ([]ConfigParamInfo, error) {
+	// Open the embedded config index file
+	logger := zerolog.Ctx(ctx)
+	logger.Info().Msg("Loading Ceph configuration parameters from embedded JSON file")
+
+	// Read the file data
+	data, err := configIndexFile.ReadFile("config-index.json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to read embedded config-index.json: %w", err)
+	}
+
+	// Parse the JSON structure (array of ConfigParamInfo objects)
+	var jsonArray []ConfigParamInfo
+	err = json.Unmarshal(data, &jsonArray)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config index JSON: %w", err)
+	}
+
+	// Check if jsonArray is sorted by Name
```

Comment:
don't need to check. since json is embedded and hadrcoded, we can check sort order in unit test